txpipe/depth/dr1.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import healpy as hp
+from ..utils.stats import ParallelStatsCalculator
+
+
+def dr1_values_iterator(data_iterator, nside, snr_threshold, snr_delta=1., mag_index=1):
+    """Loop through any input data in
+    data_iterator and computes the quantities needed for depth
+    estimation - the magnitudes of any objects that have
+    |SNR - SNR_threshold| < snr_delta.
+
+    This is passed on to the stats calculator later.
+
+    This way of writing the code makes it automatically parallel
+    and never requires loading in the full columns.
+
+    Parameters
+    ----------
+    data_iterator: iterable
+        Iterator yielding chunks of data containing ra, dec, snr and magnitude values
+
+    nside: int
+        Resolution parameter for generated map
+
+    snr_threshold: float
+        Value of SNR to use as the depth (e.g. 5.0 for 5 sigma depth)
+
+    snr_delta: float, optional
+        Half-width of the SNR values to use for the depth estimation
+
+    mag_index: int, optional
```

Comment:
I'm actually not sure what this should be!  I will open an issue to ask Erin about this - it depends on what bands were used to analyse the output.  For now I picked i because I think the hsc analysis used gri bands and wanted the r-band here.  Ideally we want to just choose the band to use.